src/OrleansProviders/Streams/Memory/MemoryStreamQueueGrain.cs

```diff
@@ -14,17 +14,13 @@ public class MemoryStreamQueueGrain : Grain, IMemoryStreamQueueGrain
         private readonly Queue<MemoryMessageData> eventQueue = new Queue<MemoryMessageData>();
         private long sequenceNumber = DateTime.UtcNow.Ticks;
 
-        /// <summary>
-        /// max event count. 
-        /// </summary>
-        private int maxEventCount = 16384;
-
         /// <summary>
         /// Enqueues an event data. If the current total count reaches the max limit. throws an exception.
         /// </summary>
         /// <param name="data"></param>
+        /// <param name="maxEventCount"></param>
         /// <returns></returns>
-        public Task Enqueue(MemoryMessageData data)
+        public Task Enqueue(MemoryMessageData data, int maxEventCount)
```

Comment:
New overload, to avoid the breaking-change.

Reply:
Addressed.